src/modules/dex/directives/createOrder/CreateOrder.js

```diff
@@ -159,20 +159,21 @@
             }
 
             setMaxPrice() {
+                if (this.price.getTokens().eq(0)) {
+                    this.amount = this.amountBalance.cloneWithTokens('0');
+                    return null;
+                }
                 if (this.priceBalance.asset.id === this.fee.asset.id) {
-                    if (this.price.getTokens().eq(0)) {
-                        this.amount = this.amountBalance.cloneWithTokens('0');
+                    const amount = this.amountBalance.cloneWithTokens(this.priceBalance.sub(this.fee)
+                        .getTokens()
+                        .div(this.price.getTokens())
+                        .round(this.amountBalance.asset.precision, BigNumber.ROUND_FLOOR));
```

Comment:
The train wreck on lines 167-170 almost literally duplicates one on lines 178-180 except having one `sub(this.fee)` call in it.
![train wreck duplication](https://user-images.githubusercontent.com/2117141/38726919-3a685b9c-3f14-11e8-9576-9992b34a0a69.png)

Reply:
You can make those lines a function and call it twice, or bring it to a discussion.